# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ashiba713/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Search Performance & Content Refresh Prioritization
I chose this lane because improving search visibility is an important challenge for businesses with large amounts of content. It is difficult for SEO teams to manually identify which pages should be updated first. By analyzing multiple search performance signals, I hope to understand how data can support better content refresh decisions and help prioritize work more effectively.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The decision this work improves is deciding which webpages should be reviewed and updated first.

The primary users are SEO specialists and content editors. They can use the ranked recommendations to prioritize their work and focus on pages most likely to benefit from content improvements.

An incorrect recommendation may waste editor time by suggesting healthy pages, or delay updates for pages that are actually declining, potentially reducing search visibility and traffic.

Machine learning is appropriate because search performance depends on many interacting signals that are difficult to combine using simple manual rules.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [13]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Pages Declining:", (df["trend_direction"] == "down").sum())
print("Median CTR:", round(df["ctr"].median(), 2))
print("Average Word Count:", round(df["word_count"].mean(), 2))

Rows: 30000
Columns: 44
Pages Declining: 16262
Median CTR: 0.07
Average Word Count: 3107.76


The starter dataset contains approximately 30,000 webpages with 44 features describing search performance and content characteristics.

The dataset contains 16,262 pages marked as declining, indicating that identifying pages requiring timely updates is an important decision-support problem.

The dataset also includes useful SEO signals such as CTR, search volume, impressions, content age, average position, and word count. These features provide enough information to explore data-driven decision support for prioritizing content refresh.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This work provides observed and data-driven insights that may help prioritize content refresh decisions.

The results should be interpreted as decision-support rather than proof of causation.

This project does not predict Google's algorithm, guarantee ranking improvements, or establish causal relationships between individual SEO factors and search performance.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.